pkg/utils/condition/condition.go

```diff
@@ -0,0 +1,102 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package condition
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"knative.dev/pkg/apis"
+)
+
+type Manager struct {
```

Comment:
Thoughts on cutting a PR to knative pkg?

Reply:
Definitely agree. I was surprised that this wasn't already part of their condition manager logic. I don't think this should be blocking for this PR but a good idea.

Reply:
Nvm, they totally already do this. I had previously read through their code but missed the part where they are handling this.